src/MetricsPHP/Stats/SlidingTimeWindowSample.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Supernova\Metric\Stats;
+
+use \MetricsPHP\Core\Clock;
+use \MetricsPHP\Helpers\MinKeyQueue;
+
+class SlidingTimeWindowSample implements Sample{
+    const COLLISION_BUFFER = 256;
+    const TRIM_THRESHOLD = 256;
+
+    protected $clock;
+    protected $measurements;
+    protected $window;
+    protected $lastTick;
+    protected $count;
+
+    public function __construct($window = 60, Clock $clock = null)
+    {
+        if (null === $clock) {
+            $clock = new Clock();
+        }
+
+        $this->setClock($clock);
+        $this->window = 1000000000 * $window * static::COLLISION_BUFFER;
+        $this->lastTick = $this->clock->getTick() * static::COLLISION_BUFFER;
+        $this->count = 0;
```

Comment:
Suggestion: replace lines 26-27 with `$this->clear();`

Otherwise this contains a bug: you don't initialize the `$this->measurements` parameter here, resulting in a fatal error when you call the `::update()` method.